### PR TITLE
Add Iterant.takeWhileWithIndex

### LINKED ISSUE #497: Add Iterant.takeWhileWithIndex
Similar with [issue #496](https://github.com/monix/monix/issues/496) we need a `takeWhileWithIndex` on `Iterant` that takes elements for as long as the given predicate returns `true`:

```scala
sealed abstract class Iterant[F[_], A] {
  // ...
  def takeWhileWithIndex(p: (A, Int) => Boolean)(implicit F: Sync[F]): Iterant[F, A]
}
```

The implementation should be very similar actually.

### PR DESCRIPTION
**Pull request by greenhat**
Fixes #497.

### CODE REVIEW

**Comment by greenhat** [comment]
I'm done. Let me know if I missed something.

File: monix-tail/shared/src/main/scala/monix/tail/Iterant.scala [hunk 1]
```diff
@@ -1392,6 +1392,24 @@ sealed abstract class Iterant[F[_], A] extends Product with Serializable {
   final def takeWhile(p: A => Boolean)(implicit F: Sync[F]): Iterant[F, A] =
     IterantTakeWhile(self, p)(F)
 
+  /** Takes longest prefix of elements zipped with their indices that satisfy the given predicate
+    * and returns a new iterant that emits those elements.
+    *
+    * Example: {{{
+    *   // Yields 1, 2
+    *   Iterant[Task].of(1, 2, 3, 4, 5, 6).takeWhile((_, idx) => idx != 2)
```
Comment (alexandru) [inline monix-tail/shared/src/main/scala/monix/tail/Iterant.scala]: An, just one very small issue with this sample, you got the name wrong: `takeWhile` => `takeWhileWithIndex`

😀

Comment (greenhat) [reply monix-tail/shared/src/main/scala/monix/tail/Iterant.scala]: @alexandru Thanks for spotting this one, Alexandru! I was so obsessed to come up with a good example that I missed it. Fixed.

**Comment by alexandru** [comment]
@greenhat I forgot; please add yourself to the `AUTHORS` file. It doesn't have to be in this PR — I'll merge this one once it turns green to get it out of the way.

**Pull request merged**